spec/models/user_spec.rb

```diff
@@ -135,6 +135,53 @@
     end
   end
 
+  describe '#validations' do
+    context 'password' do
+      let(:user) { build :user }
+
+      context 'when password is blank' do
+        before do
+          user.save
+        end
+        # password is just an attribute in user, it is not a field
+        # so, password is nil, when it is not set
+        # devise will take care of nil/blank password while registering a user
+        it 'should not throw any error' do
+          user = User.first
```

Comment:
I think we can have something in the lines of

```ruby
user = User.new
user.password = '' # invalid state
expect(user.valid?).to be false # run validations
expect(user.errors[:email]).to include("can't be blank") # check for presence of error
```
We don't need to save the record to check for it's validity

Reply:
This will not work in our case, this will still throw error from devise, and also this way it is bit confusing as we are checking errors on email, I have updated the test to use something similar and avoided saving the record.